gw_spaceheat/actors/home_alone.py

```diff
@@ -194,103 +250,125 @@ async def main(self):
+            # and now update was_onpeak for next time
+            was_onpeak = self.is_onpeak()
+
+            # Update top state
+            if self.top_state == HomeAloneTopState.Normal:
+                if self.is_onpeak() and self.is_house_cold():
+                    # implement the change in command tree. Boss: h.n -> h.onpeak-backup
+                    self.set_onpeak_backup_command_tree()
+                    # and let the normal homealone know its dormant until further notice
+                    if self.state != HomeAloneState.Dormant:
+                        self.trigger_normal_event(HomeAloneEvent.GoDormant)
+                    # trigger state change: this will also take the required actions
+                    self.enter_onpeak_backup()
+                    # Report state change to scada
+                    self.top_state_update(cause=TopStateEvent.HouseColdOnpeak)        
+            elif self.top_state == HomeAloneTopState.UsingBackupOnpeak:
+                if just_offpeak:
+                    # trigger state change: this will also take the required actions
+                    self.leave_onpeak_backup()
+                    # Report state change to scada
+                    self.top_state_update()
+                      # implement the change in command tree. Boss: h.onpeak-backup -> h.n
+                    self.set_normal_command_tree()
+                     # and let the normal homealone know its alive 
+                    if self.state == HomeAloneState.Dormant:
+                        self.trigger_normal_event(HomeAloneEvent.WakeUp)
+                                           
+            await self.check_normal_state()
 
-                time_now = datetime.now(self.timezone) 
-                if (((time_now.hour==6 or time_now.hour==16) and time_now.minute>57)
-                    or self.zone_setpoints == {}):
-                    self.get_zone_setpoints()
+            await asyncio.sleep(self.MAIN_LOOP_SLEEP_SECONDS)
 
+    async def check_normal_state(self) -> None:
+        if self.state != HomeAloneState.Dormant:                
+            previous_state = self.state
+
+            if self.is_onpeak():
+                self.storage_declared_ready = False
+                self.full_storage_energy = None
+            else:
+                self.hp_on_during_onpeak = False
+
+            time_now = datetime.now(self.timezone) 
+            if (((time_now.hour==6 or time_now.hour==16) and time_now.minute>57)
+                or self.zone_setpoints == {}):
+                self.get_zone_setpoints()
+            
+            # Require weather to move out of initializing state
+            if self.weather: 
                 self.get_latest_temperatures()
 
-                if (self.state==HomeAloneState.WaitingForTemperaturesOnPeak 
-                    or self.state==HomeAloneState.WaitingForTemperaturesOffPeak):
+                if self.state==HomeAloneState.Initializing:
```

Comment:
Tracks comparable logic in pre-oil home alone [066c0e](https://github.com/thegridelectric/gw-scada-spaceheat-python/blob/066c0eb3e830738b4a7d01e0fd4e15fdacd9c0c0/gw_spaceheat/actors/home_alone.py#L214-L238)